src/libraries/Common/tests/System/Net/EnterpriseTests/setup/README.md

```diff
@@ -0,0 +1,84 @@
+# Enterprise Scenario Testing
+
+## What Are Enterprise Scenarios?
+There are many definitions for enterprise scenarios. But generally in terms of how .NET Core networking APIs are used, enterprise scenarios are those networking scenarios that are fundamentally used by businesses (a.k.a enterprises) compared with consumers. As such, they use networking components, protocols, and security authentication mechanisms that are not used by most consumers using their home networking and Internet connections. 
+
+## Networking Components of Enterprise Scenarios
+Enterprise scenarios typically see the following kinds of components/protocols/security:
+* Although possibly connected to the Internet, most of the networking topology is internal facing. There is use of some internal “directory” service for authentication of connected computers and users. On Windows, this can include Windows Active Directory domains with computers being domain-controllers, domain-joined, or standalone computers. On Linux, this includes Kerberos realms using KDCs and participating member computers. With .NET Core being cross-platform, this now includes connections with multiple domains and realms with various cross trust between them.
+* Authentication protocols such as NTLM, Kerberos and Negotiate. These are used more than Basic and Digest. Negotiate/Kerberos requires both client and server computers to be “joined” to a common trusted directory service.
+* TLS/SSL extensively used.
+* The use of proxy servers for HTTP/HTTPS communication. These usually include authenticated proxies where the proxy server demands authentication typically with Negotiate or NTLM authentication schemes.
+* Complex DNS architectures using various A and CNAME (alias) records.
+* Use of the NegotiateStream class. This is typically seen as part of using WCF client/server architecture.
+* Impersonation/Delegation of credentials. This occurs frequently in middle-tier scenarios that involve a client computer talking with a middle-tier computer (such as a web server) which makes an outbound call to another computer such as a database server. In this case, the credentials of the client computer are delegated across the middle-tier computer so that the outbound call to the database server is made in the context of the client’s credentials.
+
+
+## Running the tests
+These tests need to be run in the dedicated Enterprise Test environment. This environment can be created on a local dev machine as long as Docker in installed. The enterprise test environment is a collection of Linux docker containers (client and servers) connected on a common docker network.
+
+Set the DOTNET_RUNTIME_REPO_ROOT environment variable to the path on your dev machine (the host machine) where the repo is installed:
+
+```
+# Windows cmd.exe shell example
+set DOTNET_RUNTIME_REPO_ROOT=s:\GitHub\runtime
```

Comment:
Why is this variable required? Given that all the scripts are already in that repo, it's pretty straightforward to infer this information.

Reply:
Many people won't look at the scripts. I wanted a set of instructions that was simple to follow. And setting this environment variable is required for the 'dev' workflow in order for the client container to build and run the tests.

Reply:
Not sure I follow. We have a lot of scripts that do this already, for example https://github.com/dotnet/runtime/blob/875a54a112999004418ca369dfb0c31a7612c3cd/eng/build.sh#L5-L13